### PR TITLE
Updates vcpkg TPL build on Windows

### PR DESCRIPTION
**Pull request by kennyweiss**
# Summary

- This is a maintenance PR
- It updates vcpkg to v2024.03.19 and the versions of the ports for blt, camp, umpire, raja, conduit and mfem.
- Umpire's external `fmt` led to a link error due to ambiguities w/ axom's internal `fmt`, so I removed `umpire` from axom's `default-features` on Windows. I think Axom will need to support and external `fmt` to resolve this.
   - Note: I tried to get vcpkg to download Umpire's `fmt` submodule, but this doesn't appear to be supported
- Also fixes an msvc compilation error in our Marching Cubes implementation related to nested macros

#### TODO:
- [x] Update RELEASE-NOTES
- [x] Ensure that this passes our manual Windows TPL build github action

### CODE REVIEW

**Comment by kennyweiss** [comment]
Passed Manual Windows TPL GitHub action -- https://github.com/LLNL/axom/actions/runs/8414360103/job/23037916412

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -19,6 +19,13 @@ The Axom project release numbers follow [Semantic Versioning](http://semver.org/
 
 ## [Unreleased] - Release date yyyy-mm-dd
 
+### Changed
+- Upgrades `vcpkg` usage for axom's automated Windows builds to its
+  [2024.03.19 release](https://github.com/microsoft/vcpkg/releases/tag/2024.03.19).
+  Also updates vcpkg port versions for axom dependencies. Temporarily removes `umpire`
+  from axom's default dependencies on Windows due to incompatibility between umpire's
+  external `fmt` and axom's vendored copy.
```
Comment (white238) [inline RELEASE-NOTES.md:27]: This is the gift that keeps on giving.. is there a path forward?

**Review by white238 — APPROVED** [review]

**Review by agcapps — APPROVED** [review]

File: RELEASE-NOTES.md (hunk 1 above)
Comment (rhornung67) [reply RELEASE-NOTES.md:27]: Probably best to excise the Axom internal copy of fmt and move to external?

**Review by rhornung67 — APPROVED** [review]
Thank you for maintaining this @kennyweiss

File: RELEASE-NOTES.md (hunk 1 above)
Comment (kennyweiss) [reply RELEASE-NOTES.md:27]: I think axom will need to support an external fmt, and this should be the default for uberenv builds.

For basic testing of axom, I think we'll want to also support an internal vendored fmt when an external one is not provided. E.g. we have some azure CI tests for windows and mac that build w/o any TPLs.

We should probably coordinate how we support this w/ Umpire and Conduit (and other RADIUSS fmt users/providers), e.g. to support header-only fmt vs. compiled fmt, and possibly other flags.

Tag: @davidbeckingsale @cyrush

**Comment by kennyweiss** [comment]
Note: Umpire + external fmt successfully built through our vcpkg setup. 

The issue I encountered was due to a conflict b/w the the external fmt and our internal one since vcpkg puts all the dlls in a common install directory.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (kennyweiss) [reply RELEASE-NOTES.md:27]: I created a ticket to track this: https://github.com/LLNL/axom/issues/1302

**Comment by cyrush** [comment]
Do you need to use fmt as a library -- could header only be sufficient? 

Same question for Umpire really.

We might be hitting pot holes when there are simplifying choices we can control.

**Pull request merged**